### DISCUSSION TITLE
How to get deletes to return an error if record doesn't exist?

### DISCUSSION THREAD

**Discussion by ghost** [sqlc-dev/sqlc#1975] [Q&A]
HI, 

Usually, when I'm deleting a single query using pgx, I will have it return an error if the record was not found. I noticed that the code throws away the rows changed tag in exec queries. Is there a way to get an error back if the delete resulted in no changes?

**Comment by mcdoker18** [comment]
Hi, you can use `: execresult` or `:execrows` instead of `:exec` to get a number of affected rows.

But if you want to get a error then maybe you can create a some wrapper for the generated`DBTX` interface which will be check the number of rows and throw a error.
